gtk/gtk_export.go

```diff
@@ -141,7 +142,9 @@ func goTreeModelForeachFunc(model *C.GtkTreeModel, path *C.GtkTreePath, iter *C.
 	treeModelForeachFuncRegistry.Unlock()
 
 	goPath := &TreePath{(*C.GtkTreePath)(path)}
+	runtime.SetFinalizer(goPath, (*TreePath).free)
 	goIter := &TreeIter{(C.GtkTreeIter)(*iter)}
+	runtime.SetFinalizer(goIter, (*TreeIter).free)
```

Comment:
Iter should not be freed, see the paragraph "The lifecycle of an iterator can be a little confusing at first..." here: https://developer.gnome.org/gtk3/stable/GtkTreeModel.html#GtkTreeModel.description

The few examples of this function I have found also don't free the treepath, and I'd tend to agree on that - we would litter finalizers for ALL treepaths in that model. Not 100% sure on this, but I assume the caller manages this reference.

Code that needs to retain that path should probably [copy it](https://developer.gnome.org/gtk3/stable/GtkTreeModel.html#gtk-tree-path-copy) which definitely should set a finalizer.

This is probably similar for `goTreeSelectionForeachFunc`

Reply:
Done